docs/core/project-sdk/msbuild-props.md

```diff
@@ -148,6 +149,15 @@ The `CopyLocalLockFileAssemblies` property is useful for plugin projects that ha
 > [!TIP]
 > Alternatively, you can use `dotnet publish` to publish the class library. For more information, see [dotnet publish](../tools/dotnet-publish.md).
 
+### PreserveCompilationContext
+
+The `PreserveCompilationContext` property causes the publish operation to copy reference assemblies for the target platform into the *ref* subdirectory of the output directory. This functionality is mostly used internally by ASP.NET, but it's also useful when you want to compile code at run time and need the reference assemblies.
```

Comment:
It would be good to get some more details on how it's used by ASP.NET.  I think it's for Razor, but not needed in all cases (ie when pages are precompiled).  Also I believe the setting does more than just save the reference assemblies, it also writes additional information about the compile options used to the deps file.

Maybe @eerhard can provide more information, or @Pilchie can route to someone who can.

Reply:
I think it would be good to have a description of how ASP.NET uses it as an example, but it shouldn't be the main description of the option.

Here's what I wrote a few years ago on the topic: https://github.com/dotnet/sdk/issues/6676#issuecomment-234981632. We could clean this up a bit here.

```suggestion
The `PreserveCompilationContext` property allows the application to compile more code at run time using the same settings that were used at build time. The assemblies referenced at build time will be copied into the *ref* subdirectory of the output directory. The names of the reference assemblies are stored in the application's `.deps.json` file along with the options passed to the compiler. This information can be retrieved using the [DependencyContext](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencymodel.dependencycontext) `CompileLibraries` and `CompilationOptions` properties.

This functionality is mostly used internally by ASP.NET MVC and Razor pages.
```

Reply:
Thanks a bunch @eerhardt, that looks great to me!

Reply:
Also tagging @pranavkm